crates/ra_hir/src/expr.rs

```diff
@@ -616,14 +641,110 @@ impl ExprCollector {
+                            let ty = if text.ends_with("isize") {
+                                UncertainIntTy::Signed(IntTy::Isize)
+                            } else if text.ends_with("i128") {
+                                UncertainIntTy::Signed(IntTy::I128)
+                            } else if text.ends_with("i64") {
+                                UncertainIntTy::Signed(IntTy::I64)
+                            } else if text.ends_with("i32") {
+                                UncertainIntTy::Signed(IntTy::I32)
+                            } else if text.ends_with("i16") {
+                                UncertainIntTy::Signed(IntTy::I16)
+                            } else if text.ends_with("i8") {
+                                UncertainIntTy::Signed(IntTy::I8)
+                            } else if text.ends_with("usize") {
+                                UncertainIntTy::Unsigned(UintTy::Usize)
+                            } else if text.ends_with("u128") {
+                                UncertainIntTy::Unsigned(UintTy::U128)
+                            } else if text.ends_with("u64") {
+                                UncertainIntTy::Unsigned(UintTy::U64)
+                            } else if text.ends_with("u32") {
+                                UncertainIntTy::Unsigned(UintTy::U32)
+                            } else if text.ends_with("u16") {
+                                UncertainIntTy::Unsigned(UintTy::U16)
+                            } else if text.ends_with("u8") {
+                                UncertainIntTy::Unsigned(UintTy::U8)
+                            } else {
+                                UncertainIntTy::Unknown
+                            };
+
+                            // TODO: actually parse integer
+                            Literal::Int(0u64, ty)
+                        }
+                        SyntaxKind::FLOAT_NUMBER => {
+                            let text = c.text().to_string();
+
+                            // FIXME: don't do it like this. maybe use something like
+                            // the IntTy::from_name functions
+                            let ty = if text.ends_with("f64") {
+                                UncertainFloatTy::Known(FloatTy::F64)
+                            } else if text.ends_with("f32") {
+                                UncertainFloatTy::Known(FloatTy::F32)
+                            } else {
+                                UncertainFloatTy::Unknown
+                            };
+
+                            // TODO: actually parse value
+                            Literal::Float(0, ty)
+                        }
+                        SyntaxKind::STRING => {
+                            // FIXME: this likely includes the " characters
+                            let text = c.text().to_string();
+                            Literal::String(text)
+                        }
+                        SyntaxKind::ARRAY_EXPR => {
```

Comment:
I don't think this can happen? Array literals result in `ARRAY_EXPR` nodes (so the `ArrayExpr` variant below)